brave/brave-ios
# [WIP] Fix#936: Add Support for OpenSearch - Auto/ Manual

This PR is implementing adding Search Engine that has [OpenSearch-Compliance](https://developer.mozilla.org/en-US/docs/Web/OpenSearch).

This is done by 

- Evaluating Open Search engine script and finding out If the website supports Open Search.
- Parse the response and retrieve the title and reference URL
- Create the URL to download the XML + also retrieve the Fav Icon
- And create Search Engine object with the data and add it to search engine list

It also includes the UX elements like add button addition on the toolbar and dialogs for approval and error handling situations

WIP: Open Search Manual Addition / Tests

## Summary of Changes

This pull request fixes #936

## Submitter Checklist:


## Test Plan:

- Go to a website that coplies to OpenSearch Compliance - Search Engines [Ecosia](https://www.ecosia.org/?c=en)
- Press Search Textfield and observe Add Button on toolbar
- Press the add button and accept adding the search engine
- Observe Engine is added in settings or while populating suggestions

## Screenshots:

![ezgif-6-92ad65bbb277](https://user-images.githubusercontent.com/6643505/102923233-66aa0d80-445d-11eb-9bd0-b24917318fee.gif)

![IMG936](https://user-images.githubusercontent.com/6643505/102924670-07013180-4460-11eb-91ba-5008ec8bda58.png)



## Reviewer Checklist:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
